### PR TITLE
clear sort column on third click and preserve sort order when paging

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Supports https://github.com/dockstore/dockstore/pull/5769
Two complicating sources of confusion,

1) on a third click the UI would clear the direction but not the sort column. 
When the no order is specified but a sort column is specified, the backend will assume the order, so the events would not return to their original order (which is sorted by date, most recent first). 

2) When sorting a column like entryname, paging would lose the sort order

**Review Instructions**
1) In the github events pages, click three times to go between the two orders and the unsorted order to return to the original state (sorted by date)
2) Sort on entryname, click between pages, may need to open network tools to see if order is preserved and matches the UI header

**Issue**
https://github.com/dockstore/dockstore/pull/5769
https://ucsc-cgl.atlassian.net/browse/SEAB-6060

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Comment by denis-yuen** [comment]
To be clear, this doesn't solve the problem but makes it more understandable what is going on

**Comment by denis-yuen** [comment]
(noticed second confusing issue in the UI)

File: src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts [hunk 1]
```diff
@@ -206,4 +222,22 @@ export class GithubAppsLogsComponent extends Base implements OnInit, AfterViewIn
       this.showContent = 'table';
     }
   }
+
+  mapSortOrder(rawSortOrder: string): 'asc' | 'desc' {
```
Comment (denis-yuen) [inline src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts:226]: I'm not entirely sure why this is needed 
Lifted from tool paging which must have had the same issue 
https://github.com/dockstore/dockstore-ui2/blob/develop/src/app/shared/tool-lister.ts#L112-L125

File: src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts [hunk 2]
```diff
@@ -125,11 +125,21 @@ export class GithubAppsLogsComponent extends Base implements OnInit, AfterViewIn
           } else {
             this.sortCol = this.sort.active;
           }
+          if (this.sort.direction == '') {
```
Comment (kathy-t) [inline src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts:128]: `==` -> `===`

**Review by kathy-t — APPROVED** [review]
I think you linked the other PR as the issue instead of the ticket

**Review by coverbeck — APPROVED** [review]

**Pull request merged**